src/WebJobs.Script/Config/ScriptHostOptionsSetup.cs

```diff
@@ -65,25 +66,26 @@ public void Configure(ScriptJobHostOptions options)
 
         private void ConfigureFunctionTimeout(IConfigurationSection jobHostSection, ScriptJobHostOptions options)
         {
+            TimeSpan functionTimeout = _environment.IsDynamic() ? DefaultFunctionTimeoutDynamic : DefaultFunctionTimeout;
             string value = jobHostSection.GetValue<string>("functionTimeout");
             if (value != null)
             {
                 TimeSpan requestedTimeout = TimeSpan.Parse(value, CultureInfo.InvariantCulture);
-
-                // Only apply limits if this is Dynamic.
-                if (_environment.IsDynamic() && (requestedTimeout < MinFunctionTimeout || requestedTimeout > MaxFunctionTimeout))
+                if (requestedTimeout < MinFunctionTimeout)
+                {
+                    string message = $"{nameof(options.FunctionTimeout)} must be greater than {MinFunctionTimeout}.";
+                    throw new ArgumentException(message);
+                }
+                // Only apply max limit if this is Dynamic.
+                if (_environment.IsDynamic() && requestedTimeout > MaxFunctionTimeoutDynamic)
                 {
-                    string message = $"{nameof(options.FunctionTimeout)} must be between {MinFunctionTimeout} and {MaxFunctionTimeout}.";
+                    string message = $"{nameof(options.FunctionTimeout)} must be less than {MaxFunctionTimeoutDynamic}.";
                     throw new ArgumentException(message);
                 }
 
-                options.FunctionTimeout = requestedTimeout;
-            }
-            else if (_environment.IsDynamic())
-            {
-                // Apply a default if this is running on Dynamic.
-                options.FunctionTimeout = DefaultFunctionTimeout;
+                functionTimeout = requestedTimeout;
             }
+            options.FunctionTimeout = functionTimeout;
 
             // TODO: DI: JobHostOptions need to me updated.
```

Comment:
@fabiocav - let me know if this TODO can be addressed now.

Reply:
Let me know what needs to be updated here. Should we set [JobHostFunctionTimeoutOptionsSetup.cs](https://github.com/Azure/azure-functions-host/blob/dev/src/WebJobs.Script/Config/JobHostFunctionTimeoutOptionsSetup.cs) here?

Reply:
This would be on the JobHostOptionsSetup instead. I think this should be done as a separate change, though.

Reply:
Ok. I will leave the TODOs :)